android/upload_sourcemap.sh

```diff
@@ -18,17 +18,55 @@ if [ ! "${INSTABUG_APP_TOKEN}" ] || [ -z "${INSTABUG_APP_TOKEN}" ]; then
+        if [ ! "${VERSION_CODE}" ] || [ -z "${VERSION_CODE}" ]; then
+            echo "versionCode could not be found, please upload the sourcemap files manually"
+            exit 0
+        fi
+    fi
+    if [ ! "${VERSION_CODE}" ] || [ -z "${VERSION_CODE}" ]; then
+         VERSION_NAME=$(grep "versionName" android/app/build.gradle | awk '{print $2}' | tr -d \''"\')
+        if [ ! "${VERSION_NAME}" ] || [ -z "${VERSION_NAME}" ]; then
+        echo "versionName could not be found, please upload the sourcemap files manually"
+        exit 0
+        fi
+    fi
+    VERSION='{"code":"'"$VERSION_CODE"'","name":"'"$VERSION_NAME"'"}'
     echo "Instabug: Token found" "\""${INSTABUG_APP_TOKEN}"\""
     echo "Instabug: Generating sourcemap files..."
-    #Generate android sourcemap
-    react-native bundle --platform android \
-    --entry-file index.js \
-    --dev false \
-    --bundle-output ./android/main.jsbundle \
-    --sourcemap-output ./android-sourcemap.json
+    IS_HERMES=$(grep "enableHermes:" ./android/app/build.gradle)
+    if [[ $IS_HERMES == *"true"* ]]; then
+        #Generate android sourcemap (HERMES)
+        react-native bundle --platform android \
+        --reset-cache \
+        --entry-file index.js \
+        --dev false \
+        --bundle-output index.android.bundle \
+        --sourcemap-output index.android.bundle.packager.map \
+
+        node_modules/hermes-engine/osx-bin/hermes -emit-binary -out index.android.bundle.hbc index.android.bundle -O -output-source-map > /dev/null 2>&1
+
+        cp index.android.bundle.hbc.map index.android.bundle.compiler.map
+
+        node node_modules/react-native/scripts/compose-source-maps.js index.android.bundle.packager.map index.android.bundle.compiler.map -o android-sourcemap.json
+        rm -rf index.android.bundle
+        rm -rf index.android.bundle.hbc.map
+        rm -rf index.android.bundle.compiler.map
+        rm -rf index.android.bundle.hbc 
+        rm -rf index.android.bundle.packager.map
+        rm -rf index.android.bundle.map
+    else
+        #Generate android sourcemap
+        react-native bundle --platform android \
+        --entry-file index.js \
+        --dev false \
+        --bundle-output ./android/main.jsbundle \
+        --sourcemap-output ./android-sourcemap.json
+    fi
     echo "Instabug: Uploading files..."
     #Upload android sourcemap
-    curl -X POST 'https://api.instabug.com/api/sdk/v3/symbols_files'  -F "symbols_file=@./android-sourcemap.json"  -F "application_token=${INSTABUG_APP_TOKEN}"  -F "platform=react_native"  -F "os=android"
-    rm -rf android-sourcemap.json
+    curl -X POST 'https://api.instabug.com/api/sdk/v3/symbols_files' -F "app_version=${VERSION}"  -F "symbols_file=@./android-sourcemap.json"  -F "application_token=${INSTABUG_APP_TOKEN}"  -F "platform=react_native"  -F "os=android"
+    # rm -rf android-sourcemap.json
     echo 
 fi
```

Comment:
Please add a newline at the end of the file, it helps with git diffing.

```suggestion
fi

```